packages/resolvers/default/src/DefaultResolver.js

```diff
@@ -18,13 +18,6 @@ export default (new Resolver({
     const resolver = new NodeResolver({
       fs: options.inputFS,
       projectRoot: options.projectRoot,
-      // Extensions are always required in URL dependencies.
-      extensions:
-        dependency.specifierType === 'commonjs' ||
-        dependency.specifierType === 'esm'
-          ? ['ts', 'tsx', 'mjs', 'js', 'jsx', 'cjs', 'json']
-          : [],
-      mainFields: ['source', 'browser', 'module', 'main'],
```

Comment:
This could be a breaking change for some plugins. Should we try to keep supporting this? Or was the plan to do a major bump on node-resolver-core anyway?

Reply:
Yeah it's a bit hard to support custom main fields with the way package.jsons are parsed now (rust discards unknown fields). Extensions could be done but idk if there are any use cases other than replicating the default logic from here. I think the safest thing would be to bump a major anyway because none of the methods other than `resolve` exist anymore, and someone might have been using them.